src/Database/Type.php

```diff
@@ -21,7 +21,7 @@
  * Encapsulates all conversion functions for values coming from database into PHP and
  * going from PHP into database.
  */
-class Type
+class Type implements TypeInterface
```

Comment:
Would that mean that a custom class both extends and implements? If the base class implements the interface, wouldn't all classes inheriting from it also implement the interface?